source/module_io/read_input_item_elec_stru.cpp

```diff
@@ -453,47 +453,35 @@ void ReadInput::item_elec_stru()
                           "set to 1, a fast algorithm is used";
         read_sync_bool(input.gamma_only);
         item.reset_value = [](const Input_Item& item, Parameter& para) {
-            Input_para& input = para.input;
-            std::string& basis_type = input.basis_type;
-            bool& gamma_only = input.gamma_only;
-            if (basis_type == "pw" && gamma_only) // pengfei Li add 2015-1-31
+            if (para.input.gamma_only && para.input.basis_type == "pw")
+            {
+                para.input.gamma_only = false;
+               GlobalV::ofs_warning << "ReadInput pw not supported in gamma_only mode."<< std::endl;
+            }
+        };
+        item.check_value = [](const Input_Item& item, const Parameter& para) {
+            if (para.inp.gamma_only && para.inp.basis_type == "pw") 
```

Comment:
If(para.inp.gamma_only && para.inp.basis_type == "pw") is useless. Since in reset_value, gamma_only is set to false.

Reply:
The line has benn delete.